framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism/block_synchronization_mechanism.spec.js

```diff
@@ -14,130 +14,238 @@
 
 'use strict';
 
-const BlockSynchronizationMechanism = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+const {
+	BlockSynchronizationMechanism,
+} = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
 const peersList = require('./peers');
+const {
+	Block: blockFixture,
+} = require('../../../../../../../mocha/fixtures/blocks');
 
-describe('BlockSynchronizationMechanism', () => {
-	const stubs = {
-		channel: {
-			invoke: jest.fn(),
-		},
-		modules: {
-			blocks: {
-				lastBlock: jest.fn(),
-			},
-		},
-	};
-	let blockSynchronizationMechanism;
-
-	beforeEach(() => {
-		blockSynchronizationMechanism = new BlockSynchronizationMechanism({
-			channel: stubs.channel,
-			modules: stubs.modules,
-		});
+describe('block_synchronization_mechanism', () => {
+	afterEach(async () => {
+		jest.clearAllMocks();
 	});
 
-	describe('#constructor', () => {});
+	describe('BlockSynchronizationMechanism', () => {
+		const activeDelegates = 101;
+		const channelMock = { invoke: jest.fn() };
 
-	describe('#_computeBestPeer', () => {
-		it('should accept an array of peers as input', () => {
-			expect(blockSynchronizationMechanism._computeBestPeer).toHaveLength(1);
+		const lastBlockGetterMock = jest.fn();
+		const blocksMock = {};
+		Object.defineProperty(blocksMock, 'lastBlock', {
+			get: lastBlockGetterMock,
 		});
 
-		it('should return a peer object', () => {
-			stubs.modules.blocks.lastBlock = {
-				height: 0,
-				prevotedConfirmedUptoHeight: 0,
-			};
-
-			const peers = [
-				{
-					lastBlockId: '12343245',
-					prevotedConfirmedUptoHeight: 1,
-					height: 67,
-					ip: '127.0.0.1',
+		const storageMock = {
+			entities: {
+				Block: {
+					getOne: jest.fn(),
 				},
-			];
+			},
+		};
+		const bftMock = {
+			finalizedHeight: null,
+		};
+		const slotsMock = {
+			getSlotNumber: jest.fn(),
+		};
+		const syncParams = {
+			channel: channelMock,
+			blocks: blocksMock,
+			storage: storageMock,
+			slots: slotsMock,
+			bft: bftMock,
+			activeDelegates,
+		};
+
+		let syncMechanism;
+
+		beforeEach(() => {
+			syncMechanism = new BlockSynchronizationMechanism(syncParams);
```

Comment:
Please use Arrange & Act & Assert comments in general, same goes for the tests below.

Reply:
The tests are very much clear and each test block is not more than 5-6 lines. No complexity to understand, I don't like such tedious job of adding these comments in each 5-6 lines of code block.

Reply:
These are opinions. That's why we voted and agreed to use them as a team. And that's why I'm trying to remind everyone this each time to create this habit. You are free to ignore my feedbacks.

Reply:
We have agreed on enforcing or making as an option after the testing standard discussion.
Therefore im dismissing this review